### PR TITLE
Support url query renaming

### PR DESCRIPTION
**Pull request by ChuntaoLu**

### CODE REVIEW

File: codegen/method.go [hunk 1]
```diff
@@ -1013,11 +1008,6 @@ func (ms *MethodSpec) setParseQueryParamStatements(
 
 		identifierName := camelCase(longQueryName) + "Query"
 
-		httpRefAnnotation := field.Annotations[antHTTPRef]
```
Comment (Raynos) [inline codegen/method.go]: this is not correct if the field.Annotation is for header then we cannot read from query params and must read from the header.

Same if the antHTTPRef annotation is body

File: codegen/test_data/clients/bar.gogen [hunk 2]
```diff
@@ -542,8 +542,14 @@ func (c *barClient) ArgWithParams(
 	// Generate full URL.
 	fullURL := c.httpClient.BaseURL + "/bar" + "/argWithParams" + "/" + string(r.UUID) + "/segment" + "/" + string(r.Params.UserUUID)
 
+	queryValues := &url.Values{}
```
Comment (Raynos) [inline codegen/test_data/clients/bar.gogen]: these values are only written into URL params and should not be written into query params.

If the http.ref annotation is params or headers then it must NOT be written into the query parameters.

File: codegen/method.go (hunk 1 above)
Comment (ChuntaoLu) [reply codegen/method.go]: Done

File: codegen/test_data/clients/bar.gogen (hunk 2 above)
Comment (ChuntaoLu) [reply codegen/test_data/clients/bar.gogen]: Done

**Pull request merged**